Add the ability to pattern match an identifier within a captures expression. (#98)
mkovaxx/anodized: crates/anodized/README.md

    ],
)]
fn add_item<T: Clone + Eq>(items: &mut Vec<T>, item: T) { todo!() }

// Pattern matching works on arrays:
#[spec(
    captures: arr as [first, second, third],
    ensures: [
        first == arr[0],
        second == arr[1],
        third == arr[2],
    ],
)]
fn match_array(arr: [i32; 3]) { todo!() }
```

- **Simple identifiers** get an automatic `old_` prefix, i.e. `x` becomes `old_x`.
- **Complex expressions** require an explicit alias using `as`, i.e. `self.items.len() as orig_len`.

> - **Pattern Match Expressions** are also possible with the `ident as pattern` syntax.

Reviewer: The goal is to support arbitrary expressions on the left side of the `as` keyword, not just a simple identifier.
```suggestion
- **Pattern Match Expressions** are also possible, e.g. `self.items as [a, b, c]`.
```
